pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -265,15 +266,53 @@ func validatePipelineResults(results []PipelineResult) (errs *apis.FieldError) {
 		expressions = filter(expressions, looksLikeResultRef)
 		resultRefs := NewResultRefs(expressions)
 		if len(expressions) != len(resultRefs) {
-			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
+			return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
 				"value").ViaFieldIndex("results", idx))
 		}
 
+		if !TaskContainsResult(result.Value, taskMap) {
+			return errs.Also(apis.ErrInvalidValue("invalid propagation of finally task results to pipeline results",
+				"value").ViaFieldIndex("results", idx))
+		}
 	}
 
 	return errs
 }
 
+// put task names in a dictionary
+func convertTaskToTaskMap(tasks []PipelineTask) map[string]string {
+	taskMap := make(map[string]string)
+	for _, val := range tasks {
+		taskMap[val.Name] = ""
+	}
+
+	return taskMap
+}
+
+// TaskContainsResult ensures the result value is referenced within the
+// task names
+func TaskContainsResult(value string, taskMap map[string]string) bool {
+	sections := splitValue(value)
+	if len(sections[0]) < 3 || len(sections[3]) < 2 {
+		return false
+	}
+
+	if len(sections) < 2 {
+		return false
+	}
+	_, ok := taskMap[sections[1]]
+	return ok
+}
+
+// split result.Value into its respective sections
+func splitValue(value string) []string {
```

Comment:
poking around some of the helper functions called in `validatePipelineResults`, it looks like the file `resultrefs.go` has some helpers for parsing results of the form `tasks.foo.results.bar`. Can you make use of some of those helpers? @chuangw6 @Yongxuanzhang or @ywluogg might know better how this works

Reply:
yeah, this [parseExpression](https://github.com/tektoncd/pipeline/blob/e7520a0a2dc78f37c3552e7f22b20d40546df6bd/pkg/apis/pipeline/v1beta1/resultref.go#L157) function can help extract taskName and resultName from the expression `tasks.<taskName>.results.<resultName>`.

Reply:
updated.